### DISCUSSION TITLE
How to read Prometheus text files?

### DISCUSSION THREAD

**Discussion by brunomrpx** [vectordotdev/vector#20362] [Q&A] [answered]
Hi guys! I've some systems that generate their metrics through Prometheus text files. The idea is to expose those metrics using the Prometheus Exporter sink. I've seen that Vector doesn't have a Prometheus File source, and the raw File source outputs just logs, so I would have to parse line by line and transform it to metrics with the respective metric type (it seems inviable). Is there any alternative? Am I missing something?

**Comment by jszwedko** [comment, marked as answer]
Hey @brunomrpx !

I don't think this is currently possible in Vector without manual parsing, but I think it is a good use-case for a new codec implementation in Vector (maybe we could call it `openmetrics`) that could be used with an existing source.
